pkg/cmd/pulumi/stack_export.go

```diff
@@ -90,6 +90,11 @@ func newStackExportCmd() *cobra.Command {
 			}
 
 			if showSecrets {
+				_, err := getStackSecretsManager(s)
+				if err != nil {
+					return errors.Wrap(err, "getting secrets manager")
+				}
```

Comment:
I would have never guessed that we have to add this call to avoid the error. Is this necessary anytime `stack.DefaultSecretsProvider` is used? Are there any other code paths where this is missing? Is there any way we can make this more explicit?

Reply:
So the issue is only regarding passphrase backend - if the environment variable isn't set, then we don't prompt for the passphrase as part of `stack.DefaultSecretsProvider`. This ultimately breaks the `stack change-secrets-provider` if there is encrypted state. For all the other backends, we usually have Env Vars set to interact with the backend - the passphrase provider doesn't need it.

Reply:
It would be great to add a comment to this explaining that we need to make this call in order to trigger the passphrase prompt.